tests/test_build/test_build.py

```diff
@@ -72,6 +72,35 @@ def test_autofunction_destructured_params(self):
             '      * **p2.foo** (*String*) --\n\n'
             '      * **p2.bar** (*String*) --\n')
 
+    def test_autofunction_defaut_values(self):
+        """Make sure params default values appear in the function definition."""
+        self._file_contents_eq(
+            'autofunction_default_values',
+            u'defaultValues(p1=42, p2="foobar")\n\n'
+            '   Arguments:\n'
+            '      * **p1** (*number*) --\n\n'
+            '      * **p2** (*string*) --\n')
+
+    def test_autofunction_defaut_values_only_in_doc(self):
+        """Make sure params default values appear in the function definition when
+        the default values only appears in JSdoc comment."""
+        self._file_contents_eq(
+            'autofunction_default_values_only_in_doc',
+            u'defaultValuesOnlyInDoc(p1=42, p2="foobar")\n\n'
+            '   Arguments:\n'
+            '      * **p1** (*number*) --\n\n'
+            '      * **p2** (*string*) --\n')
+
+    def test_autofunction_defaut_values_only_in_js(self):
+        """Make sure params default values appear in the function definition when
+        the default values only appears in Javascript Code (ES2015)."""
+        self._file_contents_eq(
+            'autofunction_default_values_only_in_js',
+            u'defaultValuesOnlyInJs(p1=42, p2=foobar)\n\n'
```

Comment:
Hmm, missing quotes here. I'm seeing what can be done about that…

Reply:
It seems to be how JSDoc works :(

it does not quote strings (the only way is to document the param including the quotes → `@param [foo="bar"]` as I did in the previous test...)

Reply:
Yes, I think the best course is to let it go without quotes. I was concerned that it would be ambiguous with default drawn from a var like…

```js
const hork = "snam!"
/**
 * @param {number} [p1=42]
 * @param {string} [p2]
 */
function defaultValues(p1, p2=hork) {}
```

…but JSDoc doesn't notice the default at all in this case. It also doesn't notice more dynamic constructs like…

```js
function defaultValues(p1, p2=woop()) {}
```

So we'll leave it unquoted for now. In fact, I can't see any way to disambiguate, say, `true` from `"true"` from the sphinx-js side. We'd have to fix it in jsdoc.

Reply:
I came to the same conclusion... The best thing to do is to document explicitly the default values and to open a ticket on jsodc repository

Reply:
I was wrong! If you feed this in…

```js
/**
 * @param p1
 * @param p2
 */
function defaultValues(p1=true, p2="true") {}
```

…it actually does the right thing:

```json
[
    {
        ...
        "params": [
            {
                "name": "p1",
                "defaultvalue": true
            },
            {
                "name": "p2",
                "defaultvalue": "true"
            }
        ],
        "name": "defaultValues",
        "longname": "defaultValues",
        "kind": "function",
        "scope": "global"
    },
    ...
]
```

Digging in more....

Reply:
So, yes, I think we can assume that any string value under `defaultvalue` is indeed meant to be a string literal. Boolean values are meant to be booleans. And ints and longs are meant to be numbers. Compound types like dicts may be possible as well. So what we really want is…

```python
params.append('%s=%s' % (name, _python_to_js(default)) if default is not None else name)
```

Where `_python_to_js` is something like…

```python
def _python_to_js(python_value):
    """Return JS code representing the given Python-equivalent data structure."""
```

I'm not going to get around to that for awhile, so feel free to give it a try if you like.

Reply:
Opened #56 to track this.